rules/linux/credential_access_bruteforce_passowrd_guessing.toml

```diff
@@ -0,0 +1,84 @@
+accounts.
+"""
+from = "now-9m"
+index = ["auditbeat-*", "logs-system.auth-*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Potential SSH Brute Force Password Guessing"
+note = """## Triage and analysis
+
+#### Possible investigation steps
+- Investigate the login failure reason code and the targeted user names.
+- Investigate the source IP address of the failed ssh login attempts.
+- Investigate other alerts associated with the user/host during the past 48 hours.
+- Identify the source and the target computer and their roles in the IT environment.
+
+### False positive analysis
+- Authentication misconfiguration or obsolete credentials.
+- Service account password expired.
+- Infrastructure or availability issue.
+
+### Response and remediation
+- Initiate the incident response process based on the outcome of the triage.
+- Isolate the involved hosts to prevent further post-compromise behavior.
+- Investigate credential exposure on systems compromised or used by the attacker to ensure all compromised accounts are
+identified. Reset passwords for these accounts and other potentially compromised credentials, such as email, business
+systems, and web services.
+- Run a full antimalware scan. This may reveal additional artifacts left in the system, persistence mechanisms, and
+malware components.
+- Determine the initial vector abused by the attacker and take action to prevent reinfection through the same vector.
+- Using the incident response data, update logging and audit policies to improve the mean time to detect (MTTD) and the
+mean time to respond (MTTR).
+"""
+risk_score = 47
+rule_id = "8cb84371-d053-4f4f-bce0-c74990e28f28"
+severity = "medium"
+tags = ["Elastic", "Host", "Linux", "Threat Detection", "Credential Access"]
+type = "eql"
+# query is sequence to 3 runs as the max authentication failure of the linux systems kick in by default after 3 runs.
+# if maxauth tries are overridden the sequence runs can be adjusted accordingly
+query = '''
+sequence by host.id, source.ip, user.name with maxspan=3s
+  [authentication where event.action  in ("ssh_login", "user_login") and
+   event.outcome == "failure" and source.ip != null and source.ip != "0.0.0.0" and source.ip != "::" ] with runs=2
+
+  [authentication where event.action  in ("ssh_login", "user_login") and
+   event.outcome == "success" and source.ip != null and source.ip != "0.0.0.0" and source.ip != "::" ]
+'''
+
+
+[[rule.threat]]
+framework = "MITRE ATT&CK"
+[[rule.threat.technique]]
+id = "T1110"
+name = "Brute Force"
+reference = "https://attack.mitre.org/techniques/T1110/"
+[[rule.threat.technique.subtechnique]]
+id = "T1110.001"
+name = "Password Guessing"
+reference = "https://attack.mitre.org/techniques/T1110/001/"
+
+[[rule.threat.technique.subtechnique]]
+id = "T1110.003"
+name = "Password Spraying"
+reference = "https://attack.mitre.org/techniques/T1110/003/"
```

Comment:
Should we remove password spraying since we are sequencing on the same user?

Reply:
We dont have a username filter, but a sequence.
This holds good if the attack is carried out for multiple user name(s) simultaneously and see which breaks. 
We would still get alerts for all targetted usernames on the host from the IP.
So its still a password spraying on the whole, not for a particular sequence.